CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 # CMakeLists to build the Spheral library.
 
-cmake_minimum_required(VERSION 3.14)
```

Comment:
Is there an explicit cmake 3.21 feature we need to bump this?

Reply:
You'll need CMake 3.21 to work with Hip and the latest BLT release.

Reply:
Probably but this PR isn't bringing in any HIP compiled code, we tried bringing this version up from 3.14 to 3.18 a while ago and had issues with BLT and CUDA. The consensus after speaking with the BLT folks was to revert back to 3.14 until we had a necessity to push the version. When HIP executing changes get pushed to Spheral is probably when we will revisit our minimum version constraint.